packages/data-context/src/actions/ErrorActions.ts

```diff
@@ -33,34 +21,34 @@ export class ErrorActions {
    */
   clearWarning (id: string) {
     this.ctx.update((d) => {
-      const warningsIndex = d.warnings.findIndex((v) => v.id === id)
+      const warningsIndex = d.diagnostics.warnings.findIndex((v) => v.id === id)
```

Comment:
Looks like this warningIndex value will be the same for `projectWarningsIndex`, `testingTypeWarningsIndex` and `appWarningsIndex` with how this was simplified. Can we clean up some of this code or am I miss-understanding?

Reply:
You are right, I just mindlessly refactored. This becomes

```ts
clearWarning (id: string) {
    this.ctx.update((d) => {
      const warningsIndex = d.diagnostics.warnings.findIndex((v) => v.id === id)

      if (warningsIndex !== -1) {
        d.diagnostics.warnings.splice(warningsIndex, 1)
      }
    })
  }
```

Reply:
done